File: codebase/core/main/src/Engine/NoopSecurityManager.fs
namespace Forest.Security

open Forest
open Forest.ComponentModel


type [<Sealed;NoComparison>] NoopSecurityManager() =
    interface ISecurityManager with
        member __.HasAccess(_ : ICommandDescriptor) = true
        member __.HasAccess(_ : ILinkDescriptor) = true
        member __.HasAccess(_ : IViewDescriptor) = true

File: codebase/core/main/src/Security/NoOpSecurityManager.cs
using Forest.ComponentModel;

namespace Forest.Security
{
    internal sealed class NoOpSecurityManager : ISecurityManager
    {
        bool ISecurityManager.HasAccess(ICommandDescriptor descriptor) => true;
        bool ISecurityManager.HasAccess(ILinkDescriptor descriptor) => true;
        bool ISecurityManager.HasAccess(IViewDescriptor descriptor) => true;
    }
}
